samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -142,6 +176,98 @@ void BLEConnectivityManager::DisconnectionHandler(bt_conn *conn, uint8_t reason)
 	}
 }
 
+#if defined(CONFIG_BT_SMP)
+void BLEConnectivityManager::SecurityChangedHandler(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	if (err) {
+		LOG_ERR("Security failed: level %d err %d", level, err);
+	} else {
+		LOG_INF("Security changed: level %d", level);
+	}
+}
+
+void BLEConnectivityManager::AuthenticationCancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_ERR("Pairing cancelled: %s\n", addr);
+
+	bt_addr_le_t a = *bt_conn_get_dst(conn);
+	Instance().RemoveBLEProvider(a);
+}
+
+void BLEConnectivityManager::PasskeyEntry(struct bt_conn *conn)
+{
+	/* Invoke callback that will forward information to user that pincode has to be provided. */
+	if (Instance().mConnectionSecurityRequestCallback) {
+		Instance().mConnectionSecurityRequestCallback(Instance().mConnectionSecurityRequestCallbackContext);
+	}
+}
+
+void BLEConnectivityManager::PairingComplete(struct bt_conn *conn, bool bonded)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_INF("Pairing completed: %s, bonded: %d\n", addr, bonded);
+
+	BLEBridgedDeviceProvider *provider = nullptr;
+
+	/* Find the created device instance based on address. */
+	for (int i = 0; i < kMaxConnectedDevices; i++) {
+		if (Instance().mConnectedProviders[i] == nullptr) {
+			continue;
+		}
+
+		bt_addr_le_t addr = Instance().mConnectedProviders[i]->GetBtAddress();
+		if (memcmp(&addr, bt_conn_get_dst(conn), sizeof(addr)) == 0) {
+			provider = Instance().mConnectedProviders[i];
+			break;
+		}
+	}
+
+	/* The device with given address was not found.  */
+	if (!provider) {
```

Comment:
Should we log error then?

Reply:
Right. Actually I started to think what should be done if it fails, because the GATT discovery will not be started and the device adding process will not be completed. On the other hand, on other failures we usually just call `RemoveBLEProvider`, but here it has no point, because provider does not exist. The question is, do we consider even realistic scenario, as provider should always be valid at this point.

Reply:
So it's up to you :)